executor/analyze.go

```diff
@@ -78,40 +78,48 @@ func (e *AnalyzeExec) Next() (*Row, error) {
 		return nil, errors.Trace(err)
 	}
 	taskCh := make(chan *analyzeTask, len(e.tasks))
-	resultCh := make(chan analyzeResult, len(e.tasks))
+	resultCh := make(chan statistics.AnalyzeResult, len(e.tasks))
 	for i := 0; i < concurrency; i++ {
 		go e.analyzeWorker(taskCh, resultCh)
 	}
 	for _, task := range e.tasks {
 		taskCh <- task
 	}
 	close(taskCh)
-	results := make([]analyzeResult, 0, len(e.tasks))
+	dom := sessionctx.GetDomain(e.ctx)
+	lease := dom.StatsHandle().Lease
+	if lease > 0 {
+		for i := 0; i < len(e.tasks); i++ {
+			result := <-resultCh
+			if result.Err != nil {
+				return nil, errors.Trace(err)
```

Comment:
If some err is nil, we can log it and continue. It's ok to return any error in line 102